### PR TITLE
Add connection timing annotations to client spans.

### LINKED ISSUE #2271: Add information from ClientConnectionTimings in BraveClient
Armeria already provides detailed connection timing information such as DNS resolution time, but BraveClient currently does not take advantage of it yet.

### PR DESCRIPTION
**Pull request by anuraaga**
Annotations are displayed in the zipkin UI as a table to be able to see time spent in individual steps of a span. It should be useful to users to be able to see client connection timings in spans.

Interestingly, until #2272 it's likely these annotations will happen before the span start time, which is not how spans are normally recorded. It shouldn't hurt though to have them since they're just numbers in a table. The other issue is actually a lot more work, so in the meantime at least this provides useful numbers.

Misc: I found that there would always be a 30-60s pause between `yarnSetup` and `yarn` on my computer and found it's because `gradle-node-plugin` adds `node_modules` as a task output, causing Gradle to hash every single file in there which takes forever. Without defining it as an output, it means if a developer mucks with `node_modules` with manual edits, `yarn` won't run again, which is arguably a good thing anyways.

Fixes #2271

### CODE REVIEW

File: brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java [hunk 1]
```diff
@@ -141,4 +163,9 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
             return delegate().execute(ctx, req);
         }
     }
+
+    private void logTiming(Span span, String prefix, long startTimeMicros, long durationNanos) {
+        span.annotate(startTimeMicros, prefix + ".start");
+        span.annotate(startTimeMicros + TimeUnit.NANOSECONDS.toMicros(durationNanos), prefix + ".end");
```
Comment (trustin) [inline brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java]: Will JVM optimize away the string concatenation here?

Comment (anuraaga) [reply brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java]: Guess not - switched to hardcoding

**Comment by trustin** [comment]
> Misc: I found that there would always be a 30-60s pause between yarnSetup and yarn on my computer and found it's because gradle-node-plugin adds node_modules as a task output, causing Gradle to hash every single file in there which takes forever. Without defining it as an output, it means if a developer mucks with node_modules with manual edits, yarn won't run again, which is arguably a good thing anyways.

Are you going to send a PR for this?

**Comment by anuraaga** [comment]
That got merged with my other PR (all my recent PRs had the patch ;) )

**Pull request merged**